### ISSUE TITLE
accuracy related query

### ISSUE THREAD

**Issue by dvatsal** [peteryuX/arcface-tf2#4]
Hi friend,

I used below config

```
# general
batch_size: 128
input_size: 112
embd_shape: 512
sub_name: 'my_arc_res50_no_central_crop'
backbone_type: 'ResNet50' # 'ResNet50', 'MobileNetV2'
head_type: ArcHead # 'ArcHead', 'NormHead'
is_ccrop: False # central-cropping or not

# train
train_dataset: './data/ms1m_bin.tfrecord'
binary_img: True
num_classes: 85742
num_samples: 5822653
epochs: 10
base_lr: 0.01
w_decay: !!float 5e-4
save_steps: 1000

# test
test_dataset: 'test_dataset'
```
At the end of training loss is 8.0511

when I tried 
is_ccrop:True 

At that time the epoch were set to 5.

The loss was near 9.x

Please suggest what can be done to improve this.

Thanks,
Vatsal

**Comment by peteryuX** [comment]
Hi friend! Your problem is why the training loss until 5 epochs is higher than until 10 epochs? The full training make better performance is right.
You can run test.py to evaluate your accuracy as the descriptions in my repo after full training, if you really need to know the accuracy is alright. The training loss is just a distance defined by objective loss function.

**Comment by dvatsal** [comment]
Thanks a lot for your guidance.

By number of epochs, I was trying to highlight that moving from 5 to 10 did not reduce the loss much.So how many epochs might be needed to have loss near 0?

So what can be done to reduce the loss further?

Yes it was my mistake in understanding the accuracy and after running `test.py` it performed evaluation and accuracy resulted like below

```
central_crop:- True

acc 0.9912, th: 1.26    (LFW)
acc 0.9393, th: 1.41    (AgeDB30)
acc 0.9210 th: 1.54     (CFP-FP)

central_crop:-  False

acc 0.9915, th: 1.26    (LFW)
acc 0.9442, th: 1.41    (AgeDB30)
acc 0.9294 th: 1.54     (CFP-FP)
```
Is there any way to achieve 0.99 accuracy on dataset with just 1 photo per class?

And how many number of  photos do you think this mechanism need to perform 0.99 accuracy?

Please advise.

Thanks again friend for your response.

**Comment by peteryuX** [comment]
You are welcome my friend, glad to hear from you. I think your accuracy results is not bad. I try my best to answer your questions as below.
1. How to reduce loss further? How many epochs can get near 0 loss?
In my opinion, the "near 0" loss might not be meaningful thing in practical situation, because training loss is a value for comparing whether your model learned the thing you want it to learn. Too many training epochs often produce overfitting when your dataset not huge and diverse enough. However, in this case, our dataset is not really small and naive, so I think maybe it just because the model converage to what it can do now. I still cannot achieve the same performance in original paper, even I try my best to follow all tricks as I known in it. I believe my implement still have something imperfects or mistakes. Generally, hardly tuning hyperparameters (learning rate, epochs, weights decay...) is one way to boast the performance. I didn't tune hyper parameters but just follow the same settings in paper, there might exist a better settings for my implement model.
2. How to achieve 99% accuracy on dataset with 1 photo per class? How many number of photos is realistic?
Actually, I am not really a expert in face recognition. It's also hard to answer many photos is enough. But It seems to me that one photo per class is too challenge, which . It should have different photos of the same person to let model learn what is the hints to identify some biometric information on face as human being known. Too small dataset cannot achieve robust model generalize to real applications.

**Comment by dvatsal** [comment]
Dear Friend.

Thanks for explaining this with such details.

It really makes sense.I would keep you posted if I achieve any better results with these items.



Thanks again.
